internal/manager/application/application.go

```diff
@@ -616,7 +632,7 @@ func (m *ApplicationManager) List(ctx context.Context, selector backend.Applicat
 
 // RevertManagedAppChanges compares the actual spec with expected spec stored in cache,
 // if actual spec doesn't match with cache, then it is reverted to be in sync with cache, which is same as principal.
-func (m *ApplicationManager) RevertManagedAppChanges(ctx context.Context, app *v1alpha1.Application) bool {
+func (m *ApplicationManager) RevertManagedAppChanges(ctx context.Context, app *v1alpha1.Application, appCache *cache.ResourceCache[v1alpha1.ApplicationSpec]) bool {
```

Comment:
For the name of this function, perhaps a name like `RevertManagedAppChangesIfNeeded`. 

(The function had this old name before your PR, but it can't hurt to tweak it now, while we're working on this part of the code!)

The `IfNeeded` suffix helps reflect that the function is called in many cases where no revert is needed. The name is a bit longer, but it makes the logic of calling functions a bit clearer IMHO. WDYT?

If you agree, we could also add `IfNeeded` to other `Revert*` functions in this PR, like `RevertRepositoryChanges`.